test/smoke.test.ts

```diff
@@ -279,30 +287,42 @@ describe("Smoke test", () => {
     it("should call undiciFetch with the correct dispatcher", async () => {
       process.env.HTTPS_PROXY = "https://127.0.0.1";
       const expectedAgent = new ProxyAgent("https://127.0.0.1");
-
-      jest.mock("../src", () => {
-        const actualModule = jest.requireActual("../src");
+      const actualModule = await import("../src/index.js");
+      jest.unstable_mockModule("../src/index.js", () => {
         return {
+          __esModule: true,
           ...actualModule,
           getProxyAgent: jest.fn(() => expectedAgent),
+          customFetch: async function (url: string, opts: any) {
+            return await undici.fetch(url, {
+              dispatcher: getProxyAgent(),
+              ...opts,
+            });
+          },
```

Comment:
I had to mock this function here, or else the test failed.

The test is failing as it's not picking up the mocked undici in `index.ts`. It seems that jest cannot mock ES imports from within a mocked ES Module.

Due to this "mock", it is not testing the code in `index.ts`

If anyone has any ideas, please throw them my way

Reply:
I ignored the `customFetch` function from the coverage, and it seems to be working just fine